portfolio/src/main/webapp/script.js

```diff
@@ -25,4 +25,19 @@ function addRandomGreeting() {
   // Add it to the page.
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
+};
+
+/**
+ * Function that adds text from DataServlet.java to the page
+ */
+async function getDataServletText() {
+    // Fetches data from DataServlet.java
+    const response = await fetch('/data');
```

Comment:
You don't have to do this in this PR, but you should try experimenting with Promise/thenables/arrow functions.

Instead of using `await`, you can do something like `fetch('/data').then(response => response.text()).then(text => document.get...innerText = text);`

Using an `async` function effectively does the same thing, but I think Promises are generally considered best practice.